### DISCUSSION TITLE
Scraping gagal terus, data dianggap tidak ada atau NoneType (Capstone DA)

### DISCUSSION THREAD

**Discussion by AGelfP** [teamalgoritma/community#653] [Q&A] [answered]
Dear Team Algoritma

Pertanyaan ini berhubungan dengan capstone DA webscraping untuk situs https://www.coingecko.com/en/coins/ethereum/historical_data?start_date=2022-01-01&end_date=2023-04-30#panel

Saat ini saya sedang berada di bagian "Finding the right key to scrap the data & Extracting the right information" pada skeleton notebook. Dari yang saya pahami, ini kan diminta memvisualisasikan pergerakan dari volume transaksi ethereum. Jadi, saya ingin mengscrap tabelnya agar nanti datanya divisualisasikan. Masalahnya,  saya udah mencoba bermacam-macam kode, contoh sebagai berikut

```
table = soup.find("div", attrs={"class":"tw-mt-5"})
table = soup.find("div", attrs={"class":"card-block"})
table = soup.find("div", attrs={"class":"coingecko-table table-responsive"})
table = soup.find("div", attrs={"class":"coingecko-table table-responsive", "data-controller" : "historical"})
table = soup.find("table", attrs={"class":"table table-striped text-sm text-lg-normal"})
table = soup.find("tbody", attrs={"data-target":"historical.tableBody"})
```

Namun hasilnya selalu sama saja
```
print(table.prettify()[1:500])

AttributeError                            Traceback (most recent call last)
Cell In[63], line 2
      1 table = soup.find("tbody", attrs={"data-target":"historical.tableBody"})
----> 2 print(table.prettify()[1:500])

AttributeError: 'NoneType' object has no attribute 'prettify'
```

Dari yang saya pahami, NoneType kan berarti datanya tidak ada. Pertanyaan saya, ini kan saya udah ambil dari inspect element. Waktu saya lihat kode-kode tersebut, yang dihighlight juga table. Kenapa tidak bisa terbaca ya?

Mohon bantuannya, terima kasih

**Comment by idoali** [comment]
Terima kasih pak atas pertanyaannya. 

Jawabannya terdapat pada issue ini pak. https://github.com/teamalgoritma/community/discussions/198 

Semoga dapat membantu

**Reply by AGelfP** [reply]
masih ngga bisa Kak, saya udh tambahin di bagian urlnya, key juga udah saya sesuaiin

```
url_get = requests.get('https://www.coingecko.com/en/coins/ethereum/historical_data#panel', headers = { 'User-Agent': 'Popular browser\'s user-agent', })

table = soup.find("table", attrs={"class":"table table-striped text-sm text-lg-normal"})
print(table.prettify()[1:500])
```

Hasilnya masih nonetype

![image](https://github.com/teamalgoritma/community/assets/122977451/b2458acd-0a9d-4a35-afc4-0e6ef635ab57)
![image](https://github.com/teamalgoritma/community/assets/122977451/b2f76e2c-eb4f-44ce-8326-5e6bb1804b7d)

**Comment by kevbow** [comment, marked as answer]
Dear @AGelfP ,

Perlu kami informasikan bahwa terdapat perubahan pada case easy dan scraping tidak bisa dilakukan pada case ini. Silakan mencoba mengerjakan case medium/hard.
Mohon maaf atas ketidaknyamanannya.

**Reply by AGelfP** [reply]
Iya Kak saya juga udah dapet informasinya di mentoring session, terima kasih ya Kak

**Comment by Charlotte-br560** [comment]
It seems like the scraping issue you're facing might be due to the table loading dynamically after the initial page load. Consider using Selenium for more robust scraping, as it can handle dynamic content. Ensure to wait for the page to fully load before scraping. Integrating [Crawlbase](https://crawlbase.com/) could also simplify the process.
